codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/protocol/HttpStringValuesMapSerializer.kt

```diff
@@ -61,17 +61,17 @@ class HttpStringValuesMapSerializer(
             val addFnName = location.addFnName
             val member = it.member
             val memberSymbol = symbolProvider.toSymbol(member)
-            when (memberTarget) {
-                is CollectionShape -> renderCollectionShape(it, memberTarget, writer)
-                is TimestampShape -> {
+            when {
+                memberTarget is CollectionShape -> renderCollectionShape(it, memberTarget, writer)
+                memberTarget is TimestampShape -> {
```

Comment:
No, it actually hampers readability somewhat. But `when` with a subject (e.g., `when (memberTarget)`) cannot be used to predicate anything but type checks (e.g., `is CollectionShape`) or equality checks (e.g. ,`== someOtherValue`). My new branch on L84 accesses an extension property (`memberTarget.isEnum`) which isn't possible when using a `when` with a subject. I'd've avoided removing the subject if I could've.

This might get better when [Kotlin 2.1 adds support for `when`-with-subject guards](https://youtrack.jetbrains.com/issue/KT-13626/Guard-conditions-in-when-with-subject).